observationtree/src/main/scala/explore/observationtree/ObsList.scala

```diff
@@ -41,19 +45,50 @@ object ObsList {
   implicit val propsReuse: Reusability[Props] = Reusability.derive
 
   def createObservation[F[_]: Applicative](
+    obsId:      Observation.Id,
     name:       String
   )(implicit c: TransactionalClient[F, ObservationDB]): F[Unit] =
     ObsQueries.ProgramCreateObservations
       .execute[F](
-        CreateObservationInput(programId = "p-2", name = name.assign)
+        CreateObservationInput(programId = "p-2", observationId = obsId.assign, name = name.assign)
       )
       .void
 
+  def deleteObservation[F[_]: Applicative](id: Observation.Id)(implicit
+    c:                                         TransactionalClient[F, ObservationDB]
+  ): F[Unit] =
+    ObsQueries.ProgramDeleteObservation.execute[F](id).void
+
   protected val component =
     ScalaComponent
       .builder[Props]
       .render_P { props =>
+        def deleteLocal(id: Observation.Id, focus: Option[ObsSummary]) =
+          props.observations.mod(l => l.filterNot(_.id === id)) *>
+            (focus match {
+              case Some(s) => props.focused.set(FocusedObs(s.id).some)
+              case None    => props.focused.set(none)
+            })
+
+        def createLocal(name: String): IO[Observation.Id] =
+          IO(Observation.Id(PosLong.unsafeFrom(Random.nextInt().abs.toLong + 1))).flatTap { oid =>
+            props.observations.mod(l =>
+              (ObsSummary(oid, name.some, observationTarget = none) :: l).reverse
```

Comment:
This will reverse the existing list. I would expect this to cause some flicker between the local change and the new list to come from the server. Maybe we can just append the new element? Both operations are `O(n)` anyway. (Or do 2 reversals)

Reply:
Won't the position of the new element depend on the sorting order? Is it always in chronological order of when they were added at the moment?

Reply:
Right we need to reverse the inner list.
@toddburnside This mimics what the server will do which is to append to the end